### PR TITLE
Remove support for ByteBuf in our transports

### PR DESCRIPTION
**Pull request by chrisvest**
Motivation:
`ByteBuf` itself has already been removed from the code base, and we're pulling in  a 4.1 dependency to satisfy those pieces of code that still need it.

Modification:
* Remove the code from the transports that was dealing with `ByteBuf`.
* Remove datagram code that was `ByteBuf` based, and reclaim the old class names for the `Buffer` based datagram classes.
* Port a few missing bits and pieces to `Buffer`, such as the String codecs.
*Disable a handful of HTTP/2 tests that could not easily fixed by inserting adaptors – fixing these will have to wait to when we port the HTTP/2 module itself to Buffer.
* `ChannelOption.RCVBUF_ALLOCATOR_USE_BUFFER` is removed as it no longer has any effect.

One thing that is _kept_ is the `alloc()` method on `Channel` and `ChannelHandlerContext`, and the related configuration/channel option stuff.
This can't be removed until we've ported HTTP/2.

Result:
ByteBuf is no longer used or supported by the transports.

### CODE REVIEW

File: testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketAutoReadTest.java [hunk 1]
```diff
@@ -15,11 +15,9 @@
  */
 package io.netty5.testsuite.transport.socket;
 
+import io.netty.buffer.Unpooled;
```
Comment (normanmaurer) [inline testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketAutoReadTest.java:18]: Kind of unexpected that this is still used.

**Review by normanmaurer — CHANGES_REQUESTED** [review]
Great work!

File: testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketAutoReadTest.java (hunk 1 above)
Comment (chrisvest) [reply testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketAutoReadTest.java:18]: Looks unused in this file. Will remove.

File: buffer/src/main/java/io/netty5/buffer/BufferUtil.java [hunk 2]
```diff
@@ -380,5 +394,20 @@ private static void appendHexDumpRowPrefix(StringBuilder dump, int row, int rowS
...
+            @Override
+            protected Buffer initialValue() throws Exception {
+                return super.initialValue();
+            }
+
+            @Override
+            protected void onRemoval(Buffer value) throws Exception {
+                super.onRemoval(value);
+            }
+        };
```
[the remark this answers is not in the document]
Comment (chrisvest) [reply buffer/src/main/java/io/netty5/buffer/BufferUtil.java:409]: Hmm… I think there was supposed to be some logic in there.

**Pull request merged**